src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -121,19 +170,29 @@ EmberAfNetworkCommissioningError OnAddWiFiNetworkCommandCallbackInternal(app::Co
     {
         if (sNetworks[i].mNetworkType == NetworkType::kUndefined)
         {
-            // Assume they are null terminated, see notes above.
-            size_t ssidLen        = strlen(reinterpret_cast<const char *>(ssid));
-            size_t credentialsLen = strlen(reinterpret_cast<const char *>(credentials));
+            VerifyOrExit(ssidLen <= sizeof(sNetworks[i].mData.mWiFi.mSSID),
+                         err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_OUT_OF_RANGE);
+            memcpy(sNetworks[i].mData.mWiFi.mSSID, ssid, ssidLen);
 
-            VerifyOrExit(ssidLen <= kMaxWiFiSSIDLen, err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_BOUNDS_EXCEEDED);
-            VerifyOrExit(credentialsLen <= kMaxWiFiCredentialsLen, err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_BOUNDS_EXCEEDED);
+            VerifyOrExit(chip::CanCastTo<std::remove_cv_t<decltype(sNetworks[i].mData.mWiFi.mSSIDLen)>>(ssidLen),
+                         err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_OUT_OF_RANGE);
+            sNetworks[i].mData.mWiFi.mSSIDLen = static_cast<std::remove_cv_t<decltype(sNetworks[i].mData.mWiFi.mSSIDLen)>>(ssidLen);
```

Comment:
Do you need the `remove_cv_t` complexity?  I'd think just using `decltype` would work?  Does it not?

In any case, this would be more readable if the type were assigned to a typename with `using whatever = type;` so the long name is not getting duplicated.